### PR TITLE
chore: Code split for policy utils

### PR DESCRIPTION
**Pull request by nidhi-nair**
## Description
Simply refactors code to allow EE overrides.

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

## Testing
Only needs existing cases to pass
- [x] JUnit
- [x] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java [hunk 1]
```diff
@@ -99,7 +99,7 @@ public class EnvManagerCEImpl implements EnvManagerCE {
     private final UserService userService;
     private final AnalyticsService analyticsService;
     private final UserRepository userRepository;
-    private final PolicyUtils policyUtils;
+    private final PolicySolution policySolution;
```
Comment (sondermanish) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java:102]: Doesn't seem like this variable getting used anywhere

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/DatasourceServiceTest.java [hunk 2]
```diff
@@ -91,7 +91,7 @@ public class DatasourceServiceTest {
     WorkspaceRepository workspaceRepository;
 
     @Autowired
-    PolicyUtils policyUtils;
+    PolicySolution policySolution;
```
Comment (sondermanish) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/DatasourceServiceTest.java:94]: this one is not getting used anywhere as well

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/ThemeServiceTest.java [hunk 3]
```diff
@@ -49,7 +49,7 @@
 public class ThemeServiceTest {
 
     @Autowired
-    PolicyUtils policyUtils;
+    PolicySolution policySolution;
```
Comment (sondermanish) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/ThemeServiceTest.java:52]: same as the above one

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/ce/ApplicationServiceCETest.java [hunk 4]
```diff
@@ -186,7 +186,7 @@ public class ApplicationServiceCETest {
     @Autowired
     PluginRepository pluginRepository;
     @Autowired
-    PolicyUtils policyUtils;
+    PolicySolution policySolution;
```
Comment (sondermanish) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/ce/ApplicationServiceCETest.java:189]: this as well

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java (hunk 1 above)
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java]: I've made the change, its inconsequential to the PR though, so I won't be running CI again

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java [hunk 5]
```diff
@@ -15,7 +15,7 @@
 import com.appsmith.server.exceptions.AppsmithException;
 import com.appsmith.server.helpers.CollectionUtils;
 import com.appsmith.server.helpers.FileUtils;
-import com.appsmith.server.helpers.PolicyUtils;
+import com.appsmith.server.solutions.PolicySolution;
```
Comment (sondermanish) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java:18]: Nit: unused import

File: app/server/appsmith-server/src/test/java/com/appsmith/server/solutions/UserSignupTest.java [hunk 6]
```diff
@@ -45,7 +44,7 @@ public class UserSignupTest {
     private ConfigService configService;
 
     @MockBean
-    private PolicyUtils policyUtils;
+    private PolicySolution policySolution;
```
Comment (sondermanish) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/solutions/UserSignupTest.java:47]: same as the above

**Review by sondermanish — APPROVED** [review]

**Pull request merged**